### PR TITLE
feat(stonedb8.0): mysql load sql with set clause should report warning. (#551)

### LINKED ISSUE #551: feature(stonedb): mysql load sql with set clause should report warning
**Is your feature request related to a problem? Please describe.**

```
CREATE TABLE `tmp_loaddata2` (
`id` int(11) NOT NULL,
`name` varchar(10) DEFAULT NULL,
`id2` int(11) NOT NULL,
PRIMARY KEY (`id`)
) ENGINE=tianmu;
```
```
echo "1,new update"   >  /data/1.txt
echo "2,new update"   >> /data/1.txt
echo "3,new update"   >> /data/1.txt
```
`LOAD DATA INFILE '/data/1.txt' Into TABLE tmp_loaddata2 FIELDS TERMINATED BY ','(id,name) set id2 =3 ;`

```
mysql> LOAD DATA INFILE '/data/1.txt' Into TABLE tmp_loaddata2 FIELDS TERMINATED BY ','(id,name) set id2 =3 ;
Query OK, 3 rows affected, 1 warning (2 min 7.31 sec)
Records: 3  Deleted: 0  Skipped: 0  Warnings: 1

mysql> show warnings;
+-------+------+----------------------------------------------------------------+
| Level | Code | Message                                                        |
+-------+------+----------------------------------------------------------------+
| Note  | 1105 | Query contains syntax that is unsupported and will be ignored. |
+-------+------+----------------------------------------------------------------+
1 row in set (0.00 sec)
```

**Describe the solution you'd like**

ommon::TIANMUError Engine::GetIOP
use stonedb8.0's THD data structure

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: close #551
warns when execute the sql as below:
`LOAD DATA INFILE '/data/1.txt' Into TABLE tmp_loaddata2 FIELDS TERMINATED BY ','(id,name) set id2 =3 ;`

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [X] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/core/engine.cpp [hunk 1]
```diff
@@ -1729,12 +1729,24 @@ common::TIANMUError Engine::GetRejectFileIOParameters(THD &thd, std::unique_ptr<
...
+  bool local_load = false;
+  uint value_list_elements = 0;
 
+  if (!for_exporter) {
+    auto cmd = down_cast<Sql_cmd_load_table *>(thd.lex->m_sql_cmd);
+    local_load = (bool)(cmd->m_is_local_file);
+    value_list_elements = cmd->m_opt_set_exprs.size();
+  }
+
+  // bool local_load = for_exporter ? false : (bool)(down_cast<Sql_cmd_load_table
```
Comment (RingsC) [inline storage/tianmu/core/engine.cpp]: These useless code can be removed now.  And, we can find it back from git log

File: storage/tianmu/core/engine.cpp [hunk 2]
```diff
@@ -1729,12 +1729,24 @@ common::TIANMUError Engine::GetRejectFileIOParameters(THD &thd, std::unique_ptr<
...
+  // *>(thd->lex->m_sql_cmd))->m_is_local_file; // stonedb8 uint value_list_elements =
+  // (bool)(down_cast<Sql_cmd_load_table *>(thd->lex->m_sql_cmd))->m_opt_set_exprs.size(); //
+  // (thd.lex)->load_value_list.elements; // stonedb8 TODO: mysql_load
+  // thr_lock_type lock_option = (thd.lex)->lock_option;
+  // stonedb8 end
   int io_mode = -1;
   char name[FN_REFLEN];
   char *tdb = 0;
```
Comment (RingsC) [inline storage/tianmu/core/engine.cpp]: using `nullptr`.

File: storage/tianmu/core/engine.cpp [hunk 3]
```diff
@@ -1729,12 +1729,24 @@ common::TIANMUError Engine::GetRejectFileIOParameters(THD &thd, std::unique_ptr<
...
+  }
+
+  // bool local_load = for_exporter ? false : (bool)(down_cast<Sql_cmd_load_table
+  // *>(thd->lex->m_sql_cmd))->m_is_local_file; // stonedb8 uint value_list_elements =
+  // (bool)(down_cast<Sql_cmd_load_table *>(thd->lex->m_sql_cmd))->m_opt_set_exprs.size(); //
+  // (thd.lex)->load_value_list.elements; // stonedb8 TODO: mysql_load
+  // thr_lock_type lock_option = (thd.lex)->lock_option;
+  // stonedb8 end
   int io_mode = -1;
   char name[FN_REFLEN];
```
Comment (RingsC) [inline storage/tianmu/core/engine.cpp]: do initialization: char name[FN_REFLEN] ={0};

File: storage/tianmu/core/engine.cpp [hunk 4]
```diff
@@ -1729,12 +1729,24 @@ common::TIANMUError Engine::GetRejectFileIOParameters(THD &thd, std::unique_ptr<
...
-  bool local_load = false; // for_exporter ? false : (bool)(thd.lex)->local_file; // stonedb8 TODO: mysql_load
-  uint value_list_elements = 0; // (thd.lex)->load_value_list.elements; // stonedb8 TODO: mysql_load
-  // thr_lock_type lock_option = (thd.lex)->lock_option;
+  // stonedb8 start
+  bool local_load = false;
+  uint value_list_elements = 0;
 
+  if (!for_exporter) {
+    auto cmd = down_cast<Sql_cmd_load_table *>(thd.lex->m_sql_cmd);
+    local_load = (bool)(cmd->m_is_local_file);
```
Comment (RingsC) [inline storage/tianmu/core/engine.cpp]: c++ type-cast style is prefer. the below code is same.

**Review by RingsC — APPROVED** [review]
LGTM

**Review by andyli029 — APPROVED** [review]
LGTM.

**Pull request merged**